site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+
+#### StatefulSet
+
+[`StatefulSet`](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) resources MAY be supported as a `backendRef`, but specifics of this integration have not yet been clarified in the full Gateway API spec.
+
+#### Multicluster support with `ServiceImport`
+
+`ServiceImport` resources allocate a virtual IP in the cluster, so MAY be allowed as a `parentRef`.
+
+`ServiceImport` would remain a valid backend option for `HTTPRoute` resources (but _not_ currently a requirement for core conformance), and could be specified alongside a `Service` `backendRef` to split traffic across clusters within a `ClusterSet` (as defined in the [Multi-cluster Service (MCS) APIs project](https://github.com/kubernetes-sigs/mcs-api)). This could be a way to solve the need described in [A use case for using Gateway APIs in Multi-Cluster](https://docs.google.com/document/d/1bjr0uAVMmEtTX4mpU_aGXXapQFsEz_Qt0OnDoVswEEI/edit).
+
+### `hostnames` field
+
+GAMMA implementations SHOULD NOT infer any functionality from the `hostnames` field on `HTTPRoute` due to current under-specification and reserved potential for future usage or API changes. For the use case of filtering incoming traffic from selected hostnames, it is recommended to guide users toward configuring [`HTTPHeaderMatch`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPHeaderMatch) rules for the [`Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) header. Functionality to be explored in future GEPs may include supporting concurrent usage of an `HTTPRoute` traffic configuration for multiple North/South `Gateways` and East/West mesh use cases or redirection of egress traffic to an in-cluster `Service`.
+
+### Limiting graph depth, preventing conflicts and cycles
+
+A `Service` may only be specified as a `parentRef` of an `HTTPRoute` if it is:
+
+* NOT already specified as a `parentRef` of another `HTTPRoute` [in the same namespace]
+    * This requirement may be relaxed later to follow the existing [precedence rules](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule), but is intended to simplify initial implementation.
+    * The namespace colocation requirement effectively forecloses using this strategy to delegate route management, but there is service mesh [user interest](https://github.com/istio/istio/issues/22997) in merging separately-defined routes together to manage large configurations, multiple versions or canary deployments for a single mesh service.
+* NOT already specified as a `backendRef` of any _other_ `HTTPRoute` with a `Service` `parentRef`
```

Comment:
Intuitively, I don't think we should add this restriction.

My rationale for that depends on the exact semantics of how routes work though. I thought we had defined it but maybe that was in my doc... we will definitely need to say what happens if there is a gateway with a route pointing to the service with route

Reply:
But this restriction only applies to another HTTPRoute with a Service parentRef (i.e. a mesh route). Therefore, it shouldn't restrict Gateway based HTTPRoutes

Reply:
```suggestion
* NOT already specified as a `backendRef` of any _other_ `HTTPRoute` with a `Service` `parentRef`
   * This restriction is intended to prevent route "inclusion/delegation" patterns or circular references from multiple tiers of routing rules _within mesh configuration_.
   * A `Service` MAY still be listed as a `backendRef` on a different `HTTPRoute` which only targets a `Gateway` as a `parentRef`. The specific behavior of how Gateways might interact with mesh routing rules will be defined in a future GEP.
```

Reply:
This is more clear, but I'll note I intend to submit a followup to try to drop this restriction.